src/Qir/Runtime/lib/QIR/OutputStream.cpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// https://stackoverflow.com/a/5419388/6362941
+// https://github.com/microsoft/qsharp-runtime/pull/511#discussion_r574170031
+// https://github.com/microsoft/qsharp-runtime/pull/511#discussion_r574194191
+
+#include <iostream>
+#include "OutputStream.hpp"
+
+namespace Microsoft           // Replace with `namespace Microsoft::Quantum` after migration to C++17.
+{
+namespace Quantum
+{
+    std::ostream* OutputStream::currentOutputStream = &std::cout; // Output to std::cout by default.
+
+    std::ostream& OutputStream::Get()
+    {
+        return *currentOutputStream;
+    }
+
+    std::ostream& OutputStream::Set(std::ostream& newOStream)
+    {
+        std::ostream& oldOStream = *currentOutputStream;
+        currentOutputStream = &newOStream;
+        return oldOStream;
```

Comment:
Is it common practice for a setter to return the old value?

Reply:
I believe in typical cases - _not_. But some standard functions do.

My code was inspired by 
* [`std::set_new_handler()`](http://www.cplusplus.com/reference/new/set_new_handler/), described in more details in chapter 8 [here](https://ptgmedia.pearsoncmg.com/images/9780321334879/samplepages/0321334876.pdf) or [here](http://wavelino.coffeecup.com/pdf/Effective%20Cshapshap.pdf)
* and (2) of [`std::ios::rdbuf()`](http://www.cplusplus.com/reference/ios/ios/rdbuf/), but this one is more a getter with an extra job of a setter ;-).